packaging/release.py

```diff
@@ -0,0 +1,1410 @@
+
+
+def set_ansible_version(current_version: Version, requested_version: Version) -> None:
+    """Set the current ansible-core version."""
+    check_ansible_version(current_version, requested_version)
+
+    if requested_version == current_version:
+        return
+
+    display.show(f"Updating version {current_version} to {requested_version} ...")
+
+    current = ANSIBLE_RELEASE_FILE.read_text()
+    updated = ANSIBLE_VERSION_PATTERN.sub(ANSIBLE_VERSION_FORMAT.format(version=requested_version), current)
+
+    if current == updated:
+        raise Exception("Failed to set the ansible-core version.")
+
+    ANSIBLE_RELEASE_FILE.write_text(updated)
+
+
+def test_built_artifact(path: pathlib.Path) -> None:
+    """Test the specified built artifact by installing it in a venv and running some basic commands."""
+    with tempfile.TemporaryDirectory() as temp_dir_name:
+        temp_dir = pathlib.Path(temp_dir_name)
+
+        venv_dir = temp_dir / "venv"
+        venv_bin_dir = venv_dir / "bin"
+
+        venv.create(venv_dir, with_pip=True)
+
+        env = os.environ.copy()
+        env.pop("PYTHONPATH", None)  # avoid interference from ansible being injected into the environment
+        env.update(
+            PATH=f"{venv_bin_dir}:{env['PATH']}",
+        )
+
+        run("pip", "install", path, *PIP_OPTIONS, env=env, cwd=CHECKOUT_DIR)
+
+        run("ansible", "--version", env=env, cwd=CHECKOUT_DIR)
+        run("ansible-test", "--version", env=env, cwd=CHECKOUT_DIR)
+
+
+def get_sdist_path(version: Version, dist_dir: pathlib.Path = DIST_DIR) -> pathlib.Path:
+    """Return the path to the sdist file."""
+    return dist_dir / f"ansible-core-{version}.tar.gz"
+
+
+def get_wheel_path(version: Version, dist_dir: pathlib.Path = DIST_DIR) -> pathlib.Path:
+    """Return the path to the wheel file."""
+    return dist_dir / f"ansible_core-{version}-py3-none-any.whl"
+
+
+def calculate_digest(path: pathlib.Path) -> str:
+    """Return the digest for the specified file."""
+    # TODO: use hashlib.file_digest once Python 3.11 is the minimum supported version
+    return hashlib.new(DIGEST_ALGORITHM, path.read_bytes()).hexdigest()
+
+
+@functools.cache
+def get_release_artifact_details(repository: str, version: Version, validate: bool) -> list[ReleaseArtifact]:
+    """Return information about the release artifacts hosted on PyPI."""
+    endpoint = PYPI_ENDPOINTS[repository]
+    url = f"{endpoint}/ansible-core/{version}/json"
+
+    opener = urllib.request.build_opener()
+    response: http.client.HTTPResponse
+
+    try:
+        with opener.open(url) as response:
+            data = json.load(response)
+    except urllib.error.HTTPError as ex:
+        if ex.status == http.HTTPStatus.NOT_FOUND:
+            raise ApplicationError(f"Version {version} not found on PyPI.") from None
+
+        raise Exception(f"Failed to get {version} from PyPI: {ex}") from ex
```

Comment:
let's try using more accurate exceptions
```suggestion
        raise RuntimeError(f"Failed to get {version} from PyPI: {ex}") from ex
```

Reply:
All uses of `raise Exception` have been replaced with `raise RuntimeError`.